src/Test.EndToEnd.Baselining/Program.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using CommandLine;
+using Test.EndToEnd.Baselining.Options;
+
+namespace Test.EndToEnd.Baselining
+{
+    /// <summary>
+    ///  Test.EndToEnd.Baselining is a console app for conducting end-to-end testing of the baselining algorithm.
+    ///  It uses sample SARIF logs, some of which contain sensitive data, so the test content is not in this repository.
+    ///  The tests are too slow to run as unit tests and are only applicable when changing the Baselining algorithm.
+    ///  
+    ///  Process
+    ///  =======
+    ///  - When the baselining algorithm is changed, run these tests to check for differences across a wide set of data.
+    ///  - Determine whether the new algorithm appears 'better' overall and investigate specific Results which don't match as they should.
+    ///  - Save the Output logs as the new baseline once you've decided to accept them.
+    /// 
+    ///  
+    ///  Content Folder Structure
+    ///  ========================
+    ///  BaselineE2E\                      [Test Root Path]
+    ///    Input\                          [SARIF log series to test]
+    ///      CloudMine-Spam\               [Series can be organized in any folder structure; using Tool\Version\ContentType\Series here]
+    ///        v1.2.12\              
+    ///          WorkItem\
+    ///            mseng\                  [A folder with files in it is an individual series]
+    ///              Baseline.sarif        [A starting Baseline before any logs are loaded]
+    ///              20191203_1530.sarif   [Each other SARIF log will be baselined in order by name; use DateTime prefixes to order them]
+    ///              20191204_1135.sarif   [...and use the file name suffix to associate them with their source]
```

Comment:
>source [](start = 105, length = 6)

Just to help me out, what does the suffix `1135` mean in this example? Is it a build number. I understand that in general it might be anything.

Reply:
In this case, it's all DateTime (yyyyMMdd_HHmm). In general the file names need to sort in order, and then can have any suffix to help map them to the source. This source is Cosmos crawls, which are keyed by DateTime, so the DateTime to the minute is all we need to find them.